zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/torrent/BittorrentNotebookRepo.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.notebook.repo.torrent;
+
+import com.frostwire.jlibtorrent.TorrentHandle;
+import com.frostwire.jlibtorrent.alerts.StateChangedAlert;
+
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileType;
+import org.apache.commons.vfs2.NameScope;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.notebook.repo.VFSNotebookRepo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+
+/**
+ * TO download note from magnet link and share a note
+ */
+public class BittorrentNotebookRepo extends VFSNotebookRepo implements TorrentEngineListener {
+  private static final Logger LOG = LoggerFactory.getLogger(BittorrentNotebookRepo.class);
+  static File torrentHomeDir;
+  static File torrentDir;
+  static File noteDownloadDir;
+  static File resumeDataDir;
+  static File settingsFile;
+  private List<TorrentSocket> connections = new CopyOnWriteArrayList<>();
+  private TorrentEngine engine;
+
+  public BittorrentNotebookRepo(ZeppelinConfiguration conf) throws IOException {
+    super(conf);
+    torrentHomeDir = new File(getRootDir().getName().getPath());
+    torrentDir = new File(torrentHomeDir.getAbsolutePath() + "/" + "torrents");
+    noteDownloadDir = new File(torrentHomeDir.getAbsolutePath() + "/" + "notes");
+    resumeDataDir = new File(torrentHomeDir.getAbsolutePath() + "/" + "resume");
+    settingsFile = new File(torrentHomeDir.getAbsolutePath() + "settings.dat");
+
+    loadLibrary();
+    setupDir();
+    engine = TorrentEngine.Instance;
+    engine.start();
+    engine.setListener(this);
+  }
+
+  private static void loadLibrary() throws IOException {
+    String path = LIBTORRENT_OS_LIBRARY_PATH();
+    if (path == null) {
+      LOG.error("Unknown Operating system");
+      throw new IOException();
```

Comment:
Let's provide the message for auser on the reason of failure though the constructor argument here